index.bs

```diff
@@ -9168,19 +9339,47 @@ a [=promise type=].
 
 <h4 id="NoInterfaceObject" extended-attribute lt="NoInterfaceObject">[NoInterfaceObject]</h4>
 
-If the [{{NoInterfaceObject}}] [=extended attribute=] appears on an [=interface=],
-it indicates that an [=interface object=]
-will not exist for the interface in the ECMAScript binding.
+<div class="advisement">
 
-<p class="advisement">
-    The [{{NoInterfaceObject}}] [=extended attribute=]
-    should not be used on [=interfaces=] that are not
-    solely used as [=supplemental interfaces=],
-    unless there are clear Web compatibility reasons for doing so. 
+    The [{{NoInterfaceObject}}] [=extended attribute=] is an undesirable feature.
+    It exists only so that legacy Web platform features can be specified.
+    It should not be used in specifications
+    unless required to specify the behavior of legacy APIs,
+    or for consistency with these APIs.
     Editors who wish to use this feature are strongly advised to discuss this
-    by <a href="https://github.com/heycam/webidl/issues/new?title=Intent%20to%20use%20[NoInterfaceObject]%20on%20a%20non-supplemental%20interface">filing an issue</a>
+    by <a href="https://github.com/heycam/webidl/issues/new?title=Intent%20to%20use%20[NoInterfaceObject]">filing an issue</a>
     before proceeding.
-</p>
+
+    <small class="non-normative">
+        The [{{NoInterfaceObject}}] [=extended attribute=] appears on the following [=interfaces=]:
+
+        {{Geolocation}},
+        {{Coordinates}},
+        {{Position}},
+        {{PositionError}},
+        {{DeviceAcceleration}},
+        {{DeviceRotationRate}},
+        {{ConstrainablePattern}},
+        {{WEBGL_compressed_texture_astc}},
+        {{WEBGL_compressed_texture_s3tc_srgb}},
+        {{WEBGL_draw_buffers}}.
+        {{WEBGL_lose_context}},
+        {{ANGLE_instanced_arrays}},
+        {{EXT_blend_minmax}},
+        {{EXT_color_buffer_float}},
+        {{EXT_disjoint_timer_query}},
+        {{OES_standard_derivatives}}, and
+        {{OES_vertex_array_object}}.
+        [[GEOLOCATION-API]]
+        [[ORIENTATION-EVENT]]
+        [[MEDIACAPTURE-STREAMS]]
+        (various [[WEBGL]] extension specifications)
+    </small>
```

Comment:
It may be worth including a note about its usage to define what-used-to-be-mixins during a transitional period, so people don't say "those aren't the only ones, DOM uses it too!".

In fact maybe it's worth including a small paragraph in the mixins section about the pattern this is replacing, for the transitional period? Not sure, hmm. Thoughts from others appreciated.

Reply:
Added a note.